client/packageutils.c

```diff
@@ -1148,6 +1148,12 @@ TDNFPopulatePkgInfos(
                       &pPkgInfo->pszLocation);
         BAIL_ON_TDNF_ERROR(dwError);
 
+        dwError = SolvGetPkgChecksumFromId(
```

Comment:
This shouldn't fail when there is no checksum available. This is why removing packages fails - it tries to get the checksum from an installed package, which isn't there.

We should check the error for `ERROR_TDNF_NO_DATA`, and in that case ignore the error. There may also be repositories without any checksum data (?), and we shouldn't fail with those (or maybe make it conditional with an option).

Later, when we want to check the checksum we can check for `pPkgInfo->pszChecksum == NULL`.